decidim-proposals/app/models/decidim/proposals/proposal.rb

```diff
@@ -89,6 +90,32 @@ def self.user_collection(author)
           .where("decidim_coauthorships.decidim_author_id = ? AND decidim_coauthorships.decidim_author_type = ? ", author.id, author.class.base_class.name)
       end
 
+      def self.retrieve_proposals_for(component)
+        Decidim::Proposals::Proposal.where(component: component).joins(:coauthorships)
+                                    .includes(:votes, :endorsements)
+                                    .where(decidim_coauthorships: { decidim_author_type: "Decidim::UserBaseEntity" })
+                                    .not_hidden
+                                    .published
+                                    .except_withdrawn
+      end
+
+      def self.newsletter_participant_ids(component)
+        proposals = retrieve_proposals_for(component).uniq
+
+        coauthors_recipients_ids = proposals.map { |p| p.notifiable_identities.pluck(:id) }.flatten.compact.uniq
+
+        ## VOTS
```

Comment:
I'd remove the comments, or if you want to add more meta-information maybe move these to methods? (In any case, it would be votes, instead of vots).